python/sparkdl/param/shared_params.py

```diff
@@ -233,3 +190,54 @@ def seKerasLoss(self, value):
 
     def getKerasLoss(self):
         return self.getOrDefault(self.kerasLoss)
+
+
+class HasOutputMapping(Params):
+    """
+    Mixin for param outputMapping: ordered list of ('outputTensorOpName', 'outputColName') pairs
+    """
+    outputMapping = Param(Params._dummy(), "outputMapping",
+                          "Mapping output :class:`tf.Operation` names to DataFrame column names",
+                          typeConverter=SparkDLTypeConverters.asTensorNameToColumnMap)
+
+    def setOutputMapping(self, value):
+        return self._set(outputMapping=value)
+
+    def getOutputMapping(self):
+        return self.getOrDefault(self.outputMapping)
+
+
+class HasInputMapping(Params):
+    """
+    Mixin for param inputMapping: ordered list of ('inputColName', 'inputTensorOpName') pairs
+    """
+    inputMapping = Param(Params._dummy(), "inputMapping",
+                         "Mapping input DataFrame column names to :class:`tf.Operation` names",
+                         typeConverter=SparkDLTypeConverters.asColumnToTensorNameMap)
+
+    def setInputMapping(self, value):
+        return self._set(inputMapping=value)
+
+    def getInputMapping(self):
+        return self.getOrDefault(self.inputMapping)
+
+
+class HasTFHParams(Params):
+    """
+    Mixin for TensorFlow model hyper-parameters
+    """
+    tfHParams = Param(Params._dummy(), "hparams",
+                      textwrap.dedent("""\
+                      instance of :class:`tf.contrib.training.HParams`, a namespace-like
+                      key-value object, storing parameters to be used to define the final
+                      TensorFlow graph for the Transformer.
+
+                      Currently accepted values are:
```

Comment:
mark as change doc => will modify as the last step

Reply:
last step of this PR or in another PR? seems like a trivial change we can just make here intead of having to keep track of it for later.

Reply:
The reason is any change takes a hour-long rebuild for Travis CI. It's reasonable to do code change first and then doc update when the code changes are accepted.

Reply:
@phi-dbq as a general philosophy, code changes should always be coming with their documentation (unless there is a specific task to track the documentation) because wrong documentation is even worse than no documentation.